util/encoding/encoding.go

```diff
@@ -599,3 +600,57 @@ func DecodeIfNull(b []byte) ([]byte, bool) {
 	}
 	return b, false
 }
+
+// EncodeTime encodes a time value, appends it to the supplied buffer,
+// and returns the final buffer. The encoding is guaranteed to be ordered
+// Such that if t1.Before(t2) then after EncodeTime(b1, t1), and
+// EncodeTime(b2, t1), Compare(b1, b2) < 0. The time zone offset is added
+// as a two byte encoding at the end of the time encoding.
+func EncodeTime(b []byte, t time.Time) []byte {
+	// Read the unix time.
+	sec := t.Unix()
+	nsec := t.Nanosecond()
+	if nsec < 0 {
+		// if it were negative we would need to change the way we encode.
+		panic("nanosecond is always non-negative")
+	}
+	b = EncodeVarint(b, sec)
+	b = EncodeVarint(b, int64(nsec))
+
+	// The offset encoding code below is copied from time.MarshalBinary()
```

Comment:
Ah, looking at the `time` package source I see that `Unix` returns a UTC value. So disregard this. Might want to add a comment explaining that since it wasn't obvious.